doc/conf.py

```diff
@@ -247,17 +246,16 @@
     'joblib': ('https://joblib.readthedocs.io/en/latest/', None),
 }
 
-if 'dev' in version:
+v = parse_version(release)
+if v.release is None:
+    raise ValueError(
+        'Ill-formed version: {!r}. Version should follow '
+        'PEP440'.format(version))
+
+if v.is_devrelease:
```

Comment:
Maybe I am missing something, but I think `pkg_resources.parse_version` does not return an object with a `.is_devrelease` attribute. On the other hand, `packaging.version.parse` does.

```py
In [1]: from packaging.version import parse                                                                                                                                        

In [2]: parse('0.22.dev0').is_devrelease                                                                                                                                           
Out[2]: True

In [3]: from pkg_resources import parse_version                                                                                                                                    

In [4]: parse_version('0.22.dev0').is_devrelease                                                                                                                                   
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-13d9ab197353> in <module>
----> 1 parse_version('0.22.dev0').is_devrelease

AttributeError: 'Version' object has no attribute 'is_devrelease'
```

Reply:
I rather not have a dependency just to parse the version. And the issue is that the new `pkg_resources` actually do have the `is_devrelease` attribute, and it works fine. The older ones however, don't. Trying to figure out when it was added to it.

Reply:
On the other hand, setuptools actually ships a version of `packaging` with it.

Reply:
:man_facepalming: `setuptools` ships `packaging=16.8`, whereas `Version().is_devrelease` was introduced in `packaging=17.0`. It worked on my system because `pkg_resources` uses the installed lib instead of the vendored one if it finds it, and I had a new `packaging `in that environment.